rs/sns/testing/src/pocket_ic.rs

```diff
@@ -0,0 +1,127 @@
+use candid::{CandidType, Encode};
+use canister_test::Wasm;
+use ic_nervous_system_integration_tests::pocket_ic_helpers::{
+    add_wasms_to_sns_wasm, install_canister_with_controllers, install_nns_suite,
+};
+use ic_nns_constants::{
+    CYCLES_MINTING_CANISTER_ID, GOVERNANCE_CANISTER_ID, IDENTITY_CANISTER_ID, LEDGER_CANISTER_ID,
+    LEDGER_INDEX_CANISTER_ID, NNS_UI_CANISTER_ID, ROOT_CANISTER_ID, SNS_AGGREGATOR_CANISTER_ID,
+    SNS_WASM_CANISTER_ID,
+};
+use pocket_ic::nonblocking::PocketIc;
+
+pub async fn bootstrap_nns(pocket_ic: &PocketIc) {
+    install_core_nns_canisters(pocket_ic).await;
+    install_frontend_nns_canisters(pocket_ic).await;
+    add_wasms_to_sns_wasm(pocket_ic, false).await.unwrap();
+}
+
+pub async fn install_core_nns_canisters(pocket_ic: &PocketIc) {
+    // TODO: use meaningful initial_balances and neuron_fund_hotkeys
+    install_nns_suite(pocket_ic, vec![], false, None, vec![], true, true).await;
+}
+
+#[derive(CandidType)]
+struct SnsAggregatorPayload {
+    pub update_interval_ms: u64,
+    pub fast_interval_ms: u64,
+}
+
+#[derive(CandidType)]
+struct NnsDappPayload {
+    args: Vec<(String, String)>,
+}
+
+pub async fn install_frontend_nns_canisters(pocket_ic: &PocketIc) {
+    let features = &[];
+    let sns_aggregator_wasm =
+        Wasm::from_location_specified_by_env_var("sns_aggregator", features).unwrap();
+    let nns_dapp_wasm = Wasm::from_location_specified_by_env_var("nns_dapp", features).unwrap();
+    let internet_identity_wasm =
+        Wasm::from_location_specified_by_env_var("internet_identity", features).unwrap();
+
+    // Refresh every second so that the NNS dapp is as up-to-date as possible
+    let nns_aggregator_payload = SnsAggregatorPayload {
+        update_interval_ms: 1000,
+        fast_interval_ms: 100,
+    };
+    install_canister_with_controllers(
+        pocket_ic,
+        "sns_aggregator",
+        SNS_AGGREGATOR_CANISTER_ID,
+        Encode!(&nns_aggregator_payload).unwrap(),
+        sns_aggregator_wasm,
+        vec![ROOT_CANISTER_ID.get(), SNS_WASM_CANISTER_ID.get()],
+    )
+    .await;
+    let internet_identity_payload: Option<()> = None;
+    install_canister_with_controllers(
+        pocket_ic,
+        "internet-identity",
+        IDENTITY_CANISTER_ID,
+        Encode!(&internet_identity_payload).unwrap(),
+        internet_identity_wasm,
+        vec![ROOT_CANISTER_ID.get()],
+    )
+    .await;
+    // TODO: make configurable
+    let endpoint = "localhost:8080";
+    let nns_dapp_payload = NnsDappPayload {
+        args: vec![
+            (String::from("API_HOST"), format!("http://{}", endpoint)),
```

Comment:
>String::from("API_HOST")

"API_HOST".to_string()` could be slightly less verbose, here as well in calls down below.